plugins/modules/system/pids.py

```diff
@@ -29,6 +38,11 @@
 - name: Printing the process IDs obtained
   ansible.builtin.debug:
     msg: "PIDS of python:{{pids_of_python.pids|join(',')}}"
+
+- name: Getting process IDs of processes matching pattern
+  community.general.pids:
+    pattern: myapp.py
```

Comment:
```suggestion
    pattern: myapp\.py
    # Without escaping the period, the period would match any character
```

Reply:
```python
regex = re.compile(re.escape(pattern), flags)
```
All values passed to the pattern option are escaped.
I chose not to allow arbitrary regex for the reasons mentioned in the summary.
If such functionality was requested I would suggest a new module which returns all process info so the user can pattern match on the controller.

Reply:
In that case, you shouldn't use `.*` as the wildcard, since that's **very** misleading.

I must admit I didn't fully understood:

> I played around with more pattern matching options, but passing user input to re is a llttle dangerous [...]

You are not passing untrusted user input to `re`, but trusted user input (i.e. what playbook/role authors pass in).

If you really want to avoid regular expressions, you shouldn't use `re` at all but just use `pattern in ...` to check whether the string can be found in the target.

Reply:
> My only problem with passing regex from users is that `re` in particular will run into issues that newer regex libraries/utilities would avoid which might confuse users. (And requiring something like `pyre2` is excessive for this)

All other places in Ansible which work with regexes have the same "problem", so using `re` is pretty much to what Ansible users are used to.

A big disadvantage of shifting the filtering to the user is that massive amounts of data have to be transferred between the remote hosts and the controller. Assume you have 10'000 servers which are running a huge amount of programs, and you want to find the PID of a specific program. Transferring 10'000 times the full process list with extra data as opposed to 10'000 times an integer is a huge difference.

Reply:
That's fair. I'll update an earlier commit which allowed regular expressions from my local with any suggestions here and push that through.